coldfront/core/utils/mou.py

```diff
@@ -0,0 +1,182 @@
+from coldfront.core.utils.common import import_from_settings
+from coldfront.core.resource.utils_.allowance_utils.computing_allowance import ComputingAllowance
+import datetime
+
+def upload_to_func(instance, filename):
+    fs = import_from_settings('FILE_STORAGE')
+    path = ''
+    type_ = ''
+    from coldfront.core.allocation.models import (AllocationAdditionRequest,
+                                                  SecureDirRequest,
+                                                  AllocationPeriod)
+    from coldfront.core.project.models import SavioProjectAllocationRequest
+    if isinstance(instance, SavioProjectAllocationRequest):
+        path += fs['details']['NEW_PROJECT_REQUEST_MOU']['location']
+        type_ = 'NewProject'
+    elif isinstance(instance, AllocationAdditionRequest):
+        path += fs['details']['SERVICE_UNITS_PURCHASE_REQUEST_MOU']['location']
+        type_ = 'AllowancePurchase'
+    elif isinstance(instance, SecureDirRequest):
+        path += fs['details']['SECURE_DIRECTORY_REQUEST_MOU']['location']
+        type_ = 'SecureDirectory'
+    path += \
+       f'{datetime.datetime.now().replace(microsecond=0).isoformat()}_' \
+       f'{instance.project.name}_{instance.requester.last_name}_{type_}_MOU.pdf'
+    return path
+
+def get_context(request_obj):
+    from coldfront.core.allocation.models import (AllocationAdditionRequest,
+                                                  SecureDirRequest,
+                                                  AllocationPeriod)
+    from coldfront.core.project.models import SavioProjectAllocationRequest
+    brc_name = 'Ken Lutz'
+    pi_name = f'{request_obj.requester.first_name} {request_obj.requester.last_name}'
+    between = f'{brc_name} (BRC) and {pi_name}'
+    date = datetime.date.today().strftime("%B %d, %Y")
+    project = request_obj.project.name
+
+    if isinstance(request_obj, SavioProjectAllocationRequest) and \
+                        ComputingAllowance(request_obj.computing_allowance) \
+                            .is_instructional():
+        service_units = request_obj.computing_allowance \
+                                                .get_attribute('Service Units')
+        course_dept = request_obj.extra_fields['course_department']
+        dept_and_pi = f'{course_dept}/{pi_name}'
+        re = f'{dept_and_pi} ICA Agreement'
+        course_name = request_obj.extra_fields['course_name']
+        point_of_contact = request_obj.extra_fields['point_of_contact']
+        allowance_end = request_obj.allocation_period.end_date
+        allowance_year = AllocationPeriod.objects.get(
```

Comment:
Instructional periods don't begin with "Allowance Year"--those are for Faculty Computing Allowances.

(Maybe it turns out that the years being computed end up being correct, but it'd make more sense to use the ICA periods.)

Reply:
What would I use instead? There's only fall/winter semester periods and 6 summer session periods